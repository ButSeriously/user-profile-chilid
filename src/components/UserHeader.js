import React from 'react';
import data from '../../storage/data.json';
import UserInfo from './UserInfo';
import UserFollowers from './UserFollowers';

class UserHeader extends React.Component {
  state = {
    likes: data.likes,
    following: data.following,
    followers: data.followers,
    isLikeAdded: false,
    isFollowersAdded: false,
  };
  toggleLike = () => {
    this.setState(prevState => ({
      likes: !this.state.isLikeAdded ?
        [prevState.likes[0], prevState.likes[1] + 1] :
        [prevState.likes[0], prevState.likes[1] - 1],
      isLikeAdded: !prevState.isLikeAdded,
    }));
  }
  toggleFollowers = () => {
    this.setState(prevState => ({
      followers: !this.state.isFollowersAdded ?
        [prevState.followers[0], prevState.followers[1] + 1] :
        [prevState.followers[0], prevState.followers[1] - 1],
      isFollowersAdded: !prevState.isFollowersAdded,
    }));
  }
  render() {
    return (
      <div className="user_header">
        <UserInfo toggleLike={this.toggleLike} />
        <UserFollowers
          likes={this.state.likes}
          following={this.state.following}
          followers={this.state.followers}
          toggleFollowers={this.toggleFollowers}
        />
      </div>
    );
  }
}

export default UserHeader;
