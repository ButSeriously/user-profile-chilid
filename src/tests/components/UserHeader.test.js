import React from 'react';
import { shallow } from 'enzyme';
import UserHeader from '../../components/UserHeader';

test('should render UserHeader', () => {
  const wrapper = shallow(<UserHeader />);
  expect(wrapper).toMatchSnapshot();
});

test('should likes toggle button', () => {
  const likes = ['Likes', 100];
  const newLikes = ['Likes', 101];
  const wrapper = shallow(<UserHeader />);
  wrapper.setState({ likes });

  // add like
  wrapper.find('UserInfo').prop('toggleLike')();
  expect(wrapper.state('likes')).toEqual(newLikes);
  expect(wrapper.state('isLikeAdded')).toBe(true);

  // subtract like
  wrapper.find('UserInfo').prop('toggleLike')();
  expect(wrapper.state('likes')).toEqual(likes);
  expect(wrapper.state('isLikeAdded')).toBe(false);
});

test('should followers toggle button', () => {
  const followers = ['Followers', 630];
  const newFollowers = ['Followers', 631];
  const wrapper = shallow(<UserHeader />);
  wrapper.setState({ followers });

  // add follower
  wrapper.find('UserFollowers').prop('toggleFollowers')();
  expect(wrapper.state('followers')).toEqual(newFollowers);
  expect(wrapper.state('isFollowersAdded')).toBe(true);

  // subtract follower
  wrapper.find('UserFollowers').prop('toggleFollowers')();
  expect(wrapper.state('followers')).toEqual(followers);
  expect(wrapper.state('isFollowersAdded')).toBe(false);
});
