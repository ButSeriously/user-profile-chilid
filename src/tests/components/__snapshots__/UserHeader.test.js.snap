// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserHeader 1`] = `
<div
  className="user_header"
>
  <UserInfo
    toggleLike={[Function]}
  />
  <UserFollowers
    followers={
      Array [
        "Followers",
        4433,
      ]
    }
    following={
      Array [
        "Following",
        723,
      ]
    }
    likes={
      Array [
        "Likes",
        121,
      ]
    }
    toggleFollowers={[Function]}
  />
</div>
`;
